src/Renderer/Label2DRenderer.js

```diff
@@ -50,16 +56,35 @@ class ScreenGrid {
 
         for (let i = minx; i <= maxx; i++) {
             for (let j = miny; j <= maxy; j++) {
-                if (this.grid[i][j]) {
-                    this.hidden.push(obj);
-                    return false;
+                if (this.grid[i][j].length > 0) {
+                    if (this.grid[i][j].some(l =>
+                        ((obj.boundaries.right > l.boundaries.left && obj.boundaries.right < l.boundaries.right)
```

Comment:
could you extract `containsBox` and `intersectsBox` methods

Reply:
Done